mitm.go

```diff
@@ -119,15 +120,20 @@ var (
 )
 
 func (p *Proxy) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	if p.Director == nil {
```

Comment:
Do we need this to be a function "global"? wouldn't make more sense inside the isWebSocket block?